services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/TableSchema.java

```diff
@@ -240,8 +241,24 @@ default T mapToItem(Map<String, AttributeValue> attributeMap, boolean preserveEm
      * @param ignoreNulls If set to true; any null values in the Java object will not be added to the output map.
      *                    If set to false; null values in the Java object will be added as {@link AttributeValue} of
      *                    type 'nul' to the output map.
+     * @param requestConfiguration denotes the level of nesting, i.e. SHALLOW or NESTED, contained in the request params based
+     * which nested update query parameters would be dealt with
      * @return A map of String to {@link AttributeValue} representing all the modelled attributes in the model object.
      */
+    Map<String, AttributeValue> itemToMap(T item, boolean ignoreNulls,
+                                          DynamoDBEnhancedRequestConfiguration requestConfiguration);
```

Comment:
Can it be just `configuration`?

Reply:
sure i can change this

Reply:
Can we move ignoreNulls to the configuration class since it's part of the configuration?

Reply:
It needs to be a default method that throws UnsupportedOperationException to be backwards compatible.

Reply:
Thats a good suggestion. I can do that